### PR TITLE
[Fix-7093][dolphincheduler-server] fix dependent node stuck in running

### LINKED ISSUE #7093: [Bug] [Dependent] stuck in running
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

dependent stuck in starting.no log output and pid is 0. It seems dependent task not launch.

### What you expected to happen

running successfully

### Anything else

_No response_

### Version

2.0.0

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

close #7093

## Brief change log

update StateWheelExecuteThread.java

### CODE REVIEW

**Comment by zhongjiajie** [comment]
@SbloodyS  I think you describe `fix issue #7093` is not work for this case.
to target specific issue from PR, we recommend use keyword `fix: #issue_id` or `close: #issue_id` or `closes: #issue_id` in your PR describe(not title, jus desc). I would not only connect issue to PR but also close issue automatically when PR is be closed.

**Comment by SbloodyS** [comment]
@zhongjiajie I'm done with the corrections.Is there anything I haven't done yet?

**Comment by zhongjiajie** [comment]
@SbloodyS As your're the first time contributor, your CI could run until committer allow it. I'm here would ask someone to help @zhuangchong @CalvinKirs

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java [hunk 1]
```diff
@@ -92,9 +92,10 @@ private void checkTask() {
         for (TaskInstance taskInstance : this.taskInstanceCheckList.values()) {
             if (TimeoutFlag.OPEN == taskInstance.getTaskDefine().getTimeoutFlag()) {
                 long timeRemain = DateUtils.getRemainTime(taskInstance.getStartTime(), taskInstance.getTaskDefine().getTimeout() * Constants.SEC_2_MINUTES_TIME_UNIT);
-                if (0 <= timeRemain && processTimeout(taskInstance)) {
+                if (0 >= timeRemain && processTimeout(taskInstance)) {
                     taskInstanceCheckList.remove(taskInstance.getId());
-                    return;
+                } else {
+                    processDependCheck(taskInstance);
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java]: maybe this else logic should be removed

Comment (SbloodyS) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java]: It's redundant...I will remove it.

**Review by caishunfeng — APPROVED** [review]
LGTM

**Comment by SbloodyS** [comment]
It seems some check failed.But I didn't see the part about my modification.What should I do? @caishunfeng

**Comment by caishunfeng** [comment]
> It seems some check failed.But I didn't see the part about my modification.What should I do? @caishunfeng

It seems timeout in e2e test. @lenboo can you retry test running?

**Comment by SbloodyS** [comment]
@caishunfeng please running test again.thanks.

**Comment by caishunfeng** [comment]
> @caishunfeng please running test again.thanks.

sorry, I don't have right to do that.  @CalvinKirs can you take a look and rerun test?

**Comment by CalvinKirs** [comment]
> > @caishunfeng please running test again.thanks.
> 
> sorry, I don't have right to do that. @CalvinKirs can you take a look and rerun test?

rerun, but I am not sure if the UT error is caused by this change.

**Review by CalvinKirs — APPROVED** [review]

**Review by lenboo — APPROVED** [review]
+1

**Pull request merged**